### PR TITLE
Feature/remove deprecated code types

### PR DESCRIPTION
**Pull request by balos1**
Adds on to remove-deprecated-code PR by removing `realtype`, `booleantype`, and related deprecated macros. We can merge this into develop once `remove-deprecated-code` is merged, but I have targeted `remove-deprecated-code` for now for diff purposes.

### CODE REVIEW

**Comment by drreynolds** [comment]
It looks like the build failed on `fsunlinsol_lapackdense_mod.c`, so you may need to rebuild the swig interfaces now that other PRs have been merged into `develop`

File: include/sundials/sundials_types_old.h
Comment (gardner48) [inline include/sundials/sundials_types_old.h:1]: I think we'd decided on naming this `sundials_types_deprecated.h`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # SUNDIALS Changelog
 
+## Changes to SUNDIALS in release 7.0.0
+
+The previously deprecated types `realtype` and `booleantype` were removed from `sundials_types.h`.
+Users should use `sunrealtype` and `sunbooleantype` instead, but the old names for the types
+can be used by including the header file `sundials_types_old.h`.
```
Comment (gardner48) [inline CHANGELOG.md:7]: I think we'd decided on naming this `sundials_types_deprecated.h`. Also slight re-wording to not suggest users use these types.
```suggestion
The previously deprecated types `realtype` and `booleantype` were removed from `sundials_types.h`
and replaced with `sunrealtype` and `sunbooleantype`. The deprecated names for these types
can be used by including the header file `sundials_types_deprecated.h` but will be fully removed in the
next major release.
```

Comment (gardner48) [reply CHANGELOG.md:7]: Also need to add this entry to the change log in the package docs

File: doc/shared/nvectors/NVector_CUDA.rst [hunk 2]
```diff
@@ -28,7 +28,7 @@ is as follows:
    struct _N_VectorContent_Cuda
    {
       sunindextype       length;
-      booleantype        own_helper;
+      sunbooleantype        own_helper;
```
Comment (drreynolds) [inline doc/shared/nvectors/NVector_CUDA.rst:31]:
```suggestion
      sunbooleantype     own_helper;
```

File: doc/shared/nvectors/NVector_HIP.rst [hunk 3]
```diff
@@ -28,7 +28,7 @@ the HIP-clang compiler. The vector content layout is as follows:
    struct _N_VectorContent_Hip
    {
       sunindextype       length;
-      booleantype        own_helper;
+      sunbooleantype        own_helper;
```
Comment (drreynolds) [inline doc/shared/nvectors/NVector_HIP.rst:31]:
```suggestion
      sunbooleantype     own_helper;
```

File: doc/shared/nvectors/NVector_SYCL.rst [hunk 4]
```diff
@@ -30,7 +30,7 @@ The vector content layout is as follows:
    struct _N_VectorContent_Sycl
    {
       sunindextype       length;
-      booleantype        own_helper;
+      sunbooleantype        own_helper;
```
Comment (drreynolds) [inline doc/shared/nvectors/NVector_SYCL.rst:33]:
```suggestion
      sunbooleantype     own_helper;
```

File: doc/shared/sunlinsol/SUNLinSol_API.rst [hunk 5]
```diff
@@ -620,13 +620,13 @@ structure is defined as
...
-     int                  (*setzeroguess)(SUNLinearSolver, booleantype);
+     int                  (*setzeroguess)(SUNLinearSolver, sunbooleantype);
      int                  (*initialize)(SUNLinearSolver);
      int                  (*setup)(SUNLinearSolver, SUNMatrix);
      int                  (*solve)(SUNLinearSolver, SUNMatrix, N_Vector,
-                                   N_Vector, realtype);
+                                   N_Vector, sunrealtype);
      int                  (*numiters)(SUNLinearSolver);
-     realtype             (*resnorm)(SUNLinearSolver);
+     sunrealtype             (*resnorm)(SUNLinearSolver);
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_API.rst:629]:
```suggestion
     sunrealtype          (*resnorm)(SUNLinearSolver);
```

File: doc/shared/sunlinsol/SUNLinSol_MagmaDense.rst [hunk 6]
```diff
@@ -109,7 +109,7 @@ The SUNLinearSolver_MagmaDense module defines the object *content* field of a
 
    struct _SUNLinearSolverContent_MagmaDense {
      int             last_flag;
-     booleantype     async;
+     sunbooleantype     async;
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_MagmaDense.rst:112]:
```suggestion
     sunbooleantype  async;
```

File: doc/shared/sunlinsol/SUNLinSol_SuperLUDIST.rst [hunk 7]
```diff
@@ -141,9 +141,9 @@ The SUNLinSol_SuperLUDIST module defines the *content* field of a
 .. code-block:: c
 
    struct _SUNLinearSolverContent_SuperLUDIST {
-     booleantype             first_factorize;
+     sunbooleantype             first_factorize;
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_SuperLUDIST.rst:144]:
```suggestion
     sunbooleantype          first_factorize;
```

File: doc/shared/sunlinsol/SUNLinSol_SuperLUDIST.rst [hunk 8]
```diff
@@ -141,9 +141,9 @@ The SUNLinSol_SuperLUDIST module defines the *content* field of a
...
      int                     last_flag;
-     realtype                berr;
+     sunrealtype                berr;
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_SuperLUDIST.rst:146]:
```suggestion
     sunrealtype             berr;
```

File: doc/shared/sunlinsol/SUNLinSol_SuperLUMT.rst [hunk 9]
```diff
@@ -133,7 +133,7 @@ The SUNLinSol_SuperLUMT module defines the *content* field of a
      sunindextype *perm_r, *perm_c;
      sunindextype N;
      int          num_threads;
-     realtype     diag_pivot_thresh;
+     sunrealtype     diag_pivot_thresh;
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_SuperLUMT.rst:136]:
```suggestion
     sunrealtype  diag_pivot_thresh;
```

File: doc/shared/sunlinsol/SUNLinSol_cuSolverSp.rst [hunk 10]
```diff
@@ -143,7 +143,7 @@ The SUNLinSol_cuSolverSp_batchQR module defines the *content* field of a
 
    struct _SUNLinearSolverContent_cuSolverSp_batchQR {
       int                last_flag;       /* last return flag                          */
-      booleantype        first_factorize; /* is this the first factorization?          */
+      sunbooleantype        first_factorize; /* is this the first factorization?          */
```
Comment (drreynolds) [inline doc/shared/sunlinsol/SUNLinSol_cuSolverSp.rst:146]:
```suggestion
      sunbooleantype     first_factorize; /* is this the first factorization?          */
```

File: doc/shared/sunmatrix/SUNMatrix_SLUNRloc.rst [hunk 11]
```diff
@@ -29,7 +29,7 @@ structure:
 .. code-block:: c
 
    struct _SUNMatrixContent_SLUNRloc {
-     booleantype   own_data;
+     sunbooleantype   own_data;
```
Comment (drreynolds) [inline doc/shared/sunmatrix/SUNMatrix_SLUNRloc.rst:32]:
```suggestion
     sunbooleantype own_data;
```

**Review by drreynolds — APPROVED** [review]

**Review by gardner48 — APPROVED** [review]

**Pull request merged**